arangod/Replication2/ReplicatedLog/LogFollower.cpp

```diff
@@ -204,13 +215,26 @@ auto replicated_log::LogFollower::appendEntries(AppendEntriesRequest req)
     }
 
     auto action = std::invoke([&]() noexcept -> DeferredAction {
+      TRI_ASSERT(req.largestCommonIndex >= self->_largestCommonIndex);
+      if (self->_largestCommonIndex > req.largestCommonIndex) {
+        LOG_CTX("fc467", TRACE, self->_follower._loggerContext)
+            << "largest common index went from " << self->_largestCommonIndex
+            << " to " << req.largestCommonIndex << ".";
```

Comment:
We first have an assertion that `req.largestCommonIndex >= self->_largestCommonIndex` and then check if `self->_largestCommonIndex > req.largestCommonIndex`, so the inverse of the previous assertion. I take it this situation is unexpected (otherwise we would not have an assertion), but then IMHO the log message should have at least WARN level. Am I missing something?

Reply:
😆  I think my intention was to log that message and check compaction if the lci became bigger. I just screwed up the condition. It should be the strict version of the assertion.